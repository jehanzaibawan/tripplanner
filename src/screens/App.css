.content-wrapper {
  margin: 2rem 0;
}

.buttons-wrapper {
  display: flex;
  width: 100%;
  margin: 1rem 0;
}
