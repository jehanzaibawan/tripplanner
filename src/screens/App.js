import React, { useState, useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";
import LinearProgress from "@material-ui/core/LinearProgress";
import TripSelector from "../components/TripSelector/TripSelector";
import ToggleGroup from "../components/ToggleGroup/ToggleGroup";
import ButtonWithIcon from "../components/ButtonWithIcon/ButtonWithIcon";
import DataList from "../components/DataList/DataList";
import { getTrips, getDestinations } from "../api/tripPlanner";
import "./App.css";

const useStyles = makeStyles({
  root: {
    flexGrow: 1
  }
});

const toggleOptions = [
  { value: "cheapest", label: "Cheapest", icon: "low_priority" },
  { value: "fastest", label: "Fastest", icon: "fast_forward" }
];

const App = props => {
  const classes = useStyles();

  const [trip, setTrip] = useState({
    selectedTripDirections: { from: "", to: "" },
    filter: toggleOptions[0].value,
    tripDirections: { departures: [], arrivals: [] }, // default data, before fetch
    trips: [],
    showLoading: false
  });

  useEffect(() => {
    const fetchData = async () => {
      const destinations = await getDestinations();

      setTrip({
        ...trip,
        tripDirections: {
          departures: destinations.departures,
          arrivals: destinations.arrivals
        }
      });
    };
    fetchData();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props]);

  const updateTripDirections = (from, to) => {
    setTrip({ ...trip, selectedTripDirections: { from, to } });
  };

  const updateFilter = filter => {
    setTrip({ ...trip, filter });
  };

  const exploreHandler = async () => {
    setTrip({ ...trip, showLoading: true });

    const trips = await getTrips({
      selectedTripDirections: trip.selectedTripDirections,
      filter: trip.filter
    });

    setTrip({ ...trip, trips, showLoading: false });
  };

  return (
    <div className={classes.root}>
      <AppBar position="static" color="default">
        <Toolbar>
          <Typography variant="h6" color="inherit">
            Trip Planner
          </Typography>
        </Toolbar>
      </AppBar>
      <Container className="content-wrapper" maxWidth="sm">
        <TripSelector
          options={trip.tripDirections}
          updateTripDirections={updateTripDirections}
        />
        <ToggleGroup
          options={toggleOptions}
          default={toggleOptions[0].value}
          updateFilter={updateFilter}
        />
        <div className="buttons-wrapper">
          <ButtonWithIcon
            disabled={
              trip.selectedTripDirections.from === "" ||
              trip.selectedTripDirections.from === "Select or type a keyword" ||
              trip.selectedTripDirections.to === "" ||
              trip.selectedTripDirections.to === "Select or type a keyword" ||
              trip.showLoading
            }
            color="primary"
            label="Explore"
            icon="find_in_page"
            onClick={exploreHandler}
          />
        </div>
        {trip.showLoading && <LinearProgress />}
        <DataList data={trip.trips} />
      </Container>
    </div>
  );
};

export default App;
