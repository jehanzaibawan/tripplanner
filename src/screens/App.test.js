import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

it("renders a snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
