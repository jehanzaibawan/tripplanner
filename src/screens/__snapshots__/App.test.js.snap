// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorDefault"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
      >
        Trip Planner
      </h6>
    </div>
  </header>
  <div
    className="MuiContainer-root content-wrapper MuiContainer-maxWidthSm"
  >
    <div
      className="makeStyles-root-79"
    />
    <div
      className="makeStyles-root-79"
    />
    <div
      className="MuiToggleButtonGroup-root"
    >
      <button
        className="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
        value="cheapest"
      >
        <span
          className="MuiToggleButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root makeStyles-leftIcon-88"
          >
            low_priority
          </span>
          <strong>
            cheapest
          </strong>
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
        value="fastest"
      >
        <span
          className="MuiToggleButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root makeStyles-leftIcon-88"
          >
            fast_forward
          </span>
          <strong>
            fastest
          </strong>
        </span>
      </button>
    </div>
    <div
      className="buttons-wrapper"
    >
      <button
        className="MuiButtonBase-root MuiButton-root makeStyles-button-108 MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root makeStyles-leftIcon-109"
          >
            find_in_page
          </span>
          Explore
        </span>
      </button>
    </div>
    <ul
      className="MuiList-root makeStyles-root-127 MuiList-padding"
    />
  </div>
</div>
`;
